rasa/core/actions/action.py

```diff
@@ -305,7 +305,15 @@ async def run(
         domain: "Domain",
     ) -> List[Event]:
         """Simple run implementation uttering a (hopefully defined) response."""
-        message = await nlg.generate(self.utter_action, tracker, output_channel.name())
+        response_ids_for_response = domain.response_ids_per_response.get(
+            self.utter_action, set()
```

Comment:
This might be a silly question, but does json payloads work well with sets? I just tried validating this json and it failed because it expects a colon (like in a dict):
```
{arguments": {"response_ids": {"1","2","3"}}}
```
Perhaps we need to convert the set to a list before passing it to `nlg.generate`.

Reply:
Oh, yes, you are right!